### PR TITLE
added the zj68 keyboard and some personal layouts for other keyboards

### PR DESCRIPTION
**Pull request by ToyVo**
## Description

the zj68 keyboard is advertised to use qmk, but it isn't in the repository, so I worked to write the code to bring it into the repo.

I've also taken the layouts for other keyboards that I have make over the last couple years and added them into this pull.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zj68/zj68.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ZJ68_H
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]: in order to clean up the include guards we have been using `#pragma once` instead. 

```suggestion
#pragma once
```

File: keyboards/zj68/zj68.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#define ZJ68_H
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]:
```suggestion

```

File: keyboards/zj68/zj68.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+#include "quantum.h"
+
+#define KEYMAP_ANSI( \
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]: The new QMK convention is to call the layout macros LAYOUT and to add an all lower case description such as 

```suggestion
#define LAYOUT_ansi( \
```

File: keyboards/zj68/zj68.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	K40, K41, K42,                   K46,       K48, K49, K4A, K4B, K4C, K4D  \
+) { \
+	{ K00,  K01,   K02,   K03,   K04,   K05,   K06,  K07,   K08,   K09,   K0A,   K0B,   K0C,  KC_NO, K0E }, \
+	{ K10,  K11,   K12,   K13,   K14,   K15,   K16,  K17,   K18,   K19,   K1A,   K1B,   K1C,  K1D,   K1E }, \
+	{ K20,  K21,   K22,   K23,   K24,   K25,   K26,  K27,   K28,   K29,   K2A,   K2B,   K2C,  KC_NO, K2E }, \
+	{ K30,  KC_NO, K32,   K33,   K34,   K35,   K36,  K37,   K38,   K39,   K3A,   K3B,   K3C,  K3D,   K3E }, \
+	{ K40,  K41,   K42,   KC_NO, KC_NO, KC_NO, K46,  KC_NO, K48,   K49,   K4A,   K4B,   K4C,  K4D,   K4E }  \
+}
+
+#define KEYMAP_ANSI_SplitBS( \
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]:
```suggestion
#define LAYOUT_ansi_split_bs( \
```

File: keyboards/zj68/zj68.h [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	K40, K41, K42,                   K46,       K48, K49, K4A, K4B, K4C, K4D  \
+) { \
+	{ K00,  K01,   K02,   K03,   K04,   K05,   K06,  K07,   K08,   K09,   K0A,   K0B,   K0C,  K0D,   K0E }, \
+	{ K10,  K11,   K12,   K13,   K14,   K15,   K16,  K17,   K18,   K19,   K1A,   K1B,   K1C,  K1D,   K1E }, \
+	{ K20,  K21,   K22,   K23,   K24,   K25,   K26,  K27,   K28,   K29,   K2A,   K2B,   K2C,  KC_NO, K2E }, \
+	{ K30,  KC_NO, K32,   K33,   K34,   K35,   K36,  K37,   K38,   K39,   K3A,   K3B,   K3C,  K3D,   K3E }, \
+	{ K40,  K41,   K42,   KC_NO, KC_NO, KC_NO, K46,  KC_NO, K48,   K49,   K4A,   K4B,   K4C,  K4D,   K4E }  \
+}
+
+#define KEYMAP_ALL( \
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]:
```suggestion
#define LAYOUT_all( \
```

File: keyboards/zj68/zj68.h [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	{ K10,  K11,   K12,   K13,   K14,   K15,   K16,  K17,   K18,   K19,   K1A,   K1B,   K1C,  K1D,   K1E }, \
+	{ K20,  K21,   K22,   K23,   K24,   K25,   K26,  K27,   K28,   K29,   K2A,   K2B,   K2C,  KC_NO, K2E }, \
+	{ K30,  K31,   K32,   K33,   K34,   K35,   K36,  K37,   K38,   K39,   K3A,   K3B,   K3C,  K3D,   K3E }, \
+	{ K40,  K41,   K42,   KC_NO, KC_NO, KC_NO, K46,  KC_NO, K48,   K49,   K4A,   K4B,   K4C,  K4D,   K4E }  \
+}
+
+
+#define LAYOUT KEYMAP_ALL
+#define KEYMAP KEYMAP_ALL
+#endif
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]:
```suggestion

```

File: keyboards/zj68/zj68.h [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+) { \
+	{ K00,  K01,   K02,   K03,   K04,   K05,   K06,  K07,   K08,   K09,   K0A,   K0B,   K0C,  K0D,   K0E }, \
+	{ K10,  K11,   K12,   K13,   K14,   K15,   K16,  K17,   K18,   K19,   K1A,   K1B,   K1C,  K1D,   K1E }, \
+	{ K20,  K21,   K22,   K23,   K24,   K25,   K26,  K27,   K28,   K29,   K2A,   K2B,   K2C,  KC_NO, K2E }, \
+	{ K30,  K31,   K32,   K33,   K34,   K35,   K36,  K37,   K38,   K39,   K3A,   K3B,   K3C,  K3D,   K3E }, \
+	{ K40,  K41,   K42,   KC_NO, KC_NO, KC_NO, K46,  KC_NO, K48,   K49,   K4A,   K4B,   K4C,  K4D,   K4E }  \
+}
+
+
+#define LAYOUT KEYMAP_ALL
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]: These aren't needed with my suggested changes
```suggestion

```

File: keyboards/zj68/zj68.h [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+#define KEYMAP KEYMAP_ALL
```
Comment (mechmerlin) [inline keyboards/zj68/zj68.h]:
```suggestion

```

File: keyboards/zj68/rules.mk [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BACKLIGHT_ENABLE ?= yes  # Enable keyboard backlight functionality
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]: We no longer use the ? as well
```suggestion
BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
```

File: keyboards/zj68/rules.mk [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+BOOTMAGIC_ENABLE ?= no	# Virtual DIP switch configuration(+1000)
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]: Also does this pcb have a reset switch? Would be useful to set this to `lite` if not or if its hard to access. 

```suggestion
BOOTMAGIC_ENABLE = no	# Virtual DIP switch configuration(+1000)
```

File: keyboards/zj68/rules.mk [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+MOUSEKEY_ENABLE ?= no	# Mouse keys(+4700)
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no	# Mouse keys(+4700)
```

File: keyboards/zj68/rules.mk [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
```

File: keyboards/zj68/rules.mk [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
CONSOLE_ENABLE = no	# Console for debug(+400)
```

File: keyboards/zj68/rules.mk [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
COMMAND_ENABLE = no    # Commands for debug and configuration
```

File: keyboards/zj68/rules.mk [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/zj68/rules.mk [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+AUDIO_ENABLE ?= yes
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
AUDIO_ENABLE = yes
```

File: keyboards/zj68/rules.mk [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```

File: keyboards/zj68/rules.mk [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+RGBLIGHT_ENABLE ?= no
```
Comment (mechmerlin) [inline keyboards/zj68/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = no
```

File: keyboards/zj68/keymaps/splitbs/keymap.c [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define _______ KC_TRNS
```
Comment (mechmerlin) [inline keyboards/zj68/keymaps/splitbs/keymap.c]: This define is already included when you `#include QMK_KEYBOARD_H`

```suggestion

```

File: keyboards/zj68/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define _______ KC_TRNS
```
Comment (mechmerlin) [inline keyboards/zj68/keymaps/default/keymap.c]: This is already included

```suggestion

```

File: keyboards/zj68/config.h [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (mechmerlin) [inline keyboards/zj68/config.h]:
```suggestion
#pragma once
```

File: keyboards/zj68/config.h [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+#define CONFIG_H
```
Comment (mechmerlin) [inline keyboards/zj68/config.h]:
```suggestion

```

File: keyboards/zj68/config.h [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 14
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+#endif
```
Comment (mechmerlin) [inline keyboards/zj68/config.h]:
```suggestion

```

File: keyboards/zj68/config.h [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+#define DEBOUNCING_DELAY 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (fauxpark) [inline keyboards/zj68/config.h]: This can be safely removed, as it's already defined in QMK and you haven't changed the default combination.

**Comment by davefp** [comment]
@ToyVo Are you still interested in working on this PR? If not I'd like to take it over as I have an interest in getting the zj68 stuff into the repo proper.

**Comment by ToyVo** [comment]
@davefp I have been meaning to get back to it, I just got an internship as a dev so I've just been busy. I will work on it this weekend and update the PR, I'm going to remove the other keymaps from the PR and implement the suggested changes. Check back in after the weekend if you don't mind.

**Comment by davefp** [comment]
@ToyVo Go for it! I built the keyboard last night so I'm messing around with the code from your branch to see what I can working in the meantime. Thanks for getting it to this stage, looking forward to seeing this PR merged :)

**Comment by ToyVo** [comment]
I made the changes that I was suggested to on this PR and I overwrote my changes on my fork to not have confusing commits. I will open up a new PR for a fresh start

**Pull request closed without merging**